File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.HashMap;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        String ss = "RTCFJMAN";
int score[] = new int[8];
        HashMap<String, Integer> point = new HashMap<>();
        
        for(String s : ss.split("")){
            point.put(s, 0);
        }
        
        for(int i=0; i<choices.length; i++){
            if(choices[i] > 4){
                point.put(survey[i].split("")[1], point.get(survey[i].split("")[1])+choices[i] -4);
            }
            if(choices[i] < 4){
                point.put(survey[i].split("")[0], point.get(survey[i].split("")[0])+4-choices[i]);
            }
        }
        
        for(int i=0; i<score.length; i++){
            score[i] = point.get(ss.split("")[i]);
        }
        
        
        for(int i=0; i<score.length; i+=2){
            if(score[i] >= score[i+1]){
                answer += ss.split("")[i];
            }
            else answer += ss.split("")[i+1];
        }
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    const personality = {'RT': [0, 0], "CF": [0, 0], "JM": [0, 0], "AN": [0, 0]}
    const n = survey.length;
    
    for (let i = 0; i < survey.length; i++) {
        if (choices[i] === 4) continue;
        
        if (personality[survey[i]]) {
            if (choices[i] < 4) {
                personality[survey[i]][0] += 4 - choices[i];
            } else {
                personality[survey[i]][1] += choices[i] - 4;
            }
        } else {
            const reverse = `${survey[i][1]}${survey[i][0]}`;
            if (choices[i] < 4) {
                personality[reverse][1] += 4 - choices[i];
            } else {
                personality[reverse][0] += choices[i] - 4;
            }
        }
    }
    
    let answer = '';
    for (const key in personality) {
        if (personality[key][0] >= personality[key][1]) {
            answer += key[0];
        } else {
            answer += key[1];
        }
    }
    
    return answer;
}